zerver/tests/test_link_embed.py

```diff
@@ -374,3 +375,116 @@ def test_link_embed_data_from_cache(self) -> None:
             key = preview_url_cache_key(url)
             cache_set(key, link_embed_data, 'database')
             self.assertEqual(link_embed_data, link_embed_data_from_cache(url))
+
+    @override_settings(INLINE_URL_EMBED_PREVIEW=True)
+    def test_link_preview_non_html_data(self) -> None:
+        email = self.example_email('hamlet')
+        self.login(email)
+        url = 'http://test.org/audio.mp3'
+        with mock.patch('zerver.lib.actions.queue_json_publish') as patched:
+            msg_id = self.send_stream_message(email, "Scotland", topic_name="foo", content=url)
+            patched.assert_called_once()
+            queue = patched.call_args[0][0]
+            self.assertEqual(queue, "embed_links")
+            event = patched.call_args[0][1]
+
+        headers = {'content-type': 'application/octet-stream'}
+        mocked_response = mock.Mock(side_effect=self.create_mock_response(url, headers=headers))
+
+        with self.settings(TEST_SUITE=False, CACHES=TEST_CACHES):
```

Comment:
I'd also like to get away from the pattern of `self.settings(TEST_SUITE=False)`; that's definitely a land mine waiting to blow up on us.  I don't remember why we had it.

Reply:
I'll dig around and see if I can figure out why we have this pattern. :+1: